src/main/java/io/jenkins/plugins/folderauth/FolderAuthorizationStrategyManagementLink.java

```diff
@@ -443,4 +446,131 @@ public void doRemoveSidFromAgentRole(@QueryParameter(required = true) String rol
         FolderAuthorizationStrategyAPI.removeSidFromAgentRole(sid, roleName);
         redirect();
     }
+
+    /**
+     * API Method to get an {@link AgentRole}
+     * Example: {@code curl -X GET 'http://localhost:8080/jenkins/folder-auth/getAgentRole?name=agentSmithRole}
+     *
+     * @param name name of the role (single, no list)
+     */
+    @Restricted(NoExternalUse.class)
+    public void doGetAgentRole(@QueryParameter(required = true) String name) throws IOException {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        JSONObject responseJson = new JSONObject();
+        AgentRole role = FolderAuthorizationStrategyAPI.getAgentRole(name);
+        if (role != null) {
+            responseJson.put("name", role.getName());
+            responseJson.put("agents", role.getAgents());
+            responseJson.put("sids", role.getSids());
+            responseJson.put("permissions", getPermissionsFromRole(role));
+        }
+        Stapler.getCurrentResponse().setContentType("application/json;charset=UTF-8");
```

Comment:
I think it is possible to directly return a `JSONObject` from a `doFoo` function like done here: https://github.com/jenkinsci/folder-auth-plugin/blob/5fc71be89c0560e09f6d03c3817c0738b8fb3fff/src/main/java/io/jenkins/plugins/folderauth/FolderAuthorizationStrategyManagementLink.java#L274

Reply:
Oh I wasn't aware of that. Is that due to the `@GET` annotation from stapler?

Reply:
`@GET` is to restrict the call to only GET requests: https://github.com/stapler/stapler/blob/master/core/src/main/java/org/kohsuke/stapler/verb/GET.java

I think returning a JSON response is implemented here: https://github.com/stapler/stapler/blob/0a23a9000de1dafb6a16fa883504697443cc05e3/core/src/main/java/org/kohsuke/stapler/json/JsonHttpResponse.java